packages/mobile/src/recipients/reducer.ts

```diff
@@ -12,44 +12,54 @@ export interface State {
   // valoraRecipientCache contains accounts that the user has sent/recieved transactions from,
   // and includes CIP8 profile data if available
   valoraRecipientCache: AddressToRecipient
+  rewardsSenders: string[]
 }
 
-const initialState = {
+const initialState: State = {
   phoneRecipientCache: {},
   valoraRecipientCache: {},
+  rewardsSenders: [],
 }
 
-export const recipientsReducer = (
-  state: State | undefined = initialState,
-  action: ActionTypes | RehydrateAction
-) => {
-  switch (action.type) {
-    case REHYDRATE: {
+const rehydrate = createAction<any>(REHYDRATE)
+export const setPhoneRecipientCache = createAction<NumberToRecipient>(
+  'RECIPIENTS/SET_PHONE_RECIPIENT_CACHE'
+)
+export const updateValoraRecipientCache = createAction<AddressToRecipient>(
+  'RECIPIENTS/SET_VALORA_RECIPIENT_CACHE'
+)
+export const setRewardsSenders = createAction<string[]>('RECIPIENTS/SET_REWARDS_SENDERS')
```

Comment:
We should avoid using setter actions and [model them as events](https://redux.js.org/style-guide/style-guide#model-actions-as-events-not-setters).